spatialmath/mesh.go

```diff
@@ -539,3 +544,56 @@ func calculatePolygonAreaWithTriangulation(vertices []r3.Vector) float64 {
 		return totalArea
 	}
 }
+
+// trianglesToPLYBytes converts the mesh's triangles to bytes in PLY format.
+func (m *Mesh) trianglesToPLYBytes() []byte {
+	// Collect all unique vertices and create vertex-to-index mapping
+	vertexMap := make(map[string]int)
+	vertices := make([]r3.Vector, 0)
+
+	for _, tri := range m.triangles {
+		for _, pt := range tri.Points() {
+			scaledPt := r3.Vector{X: pt.X / 1000.0, Y: pt.Y / 1000.0, Z: pt.Z / 1000.0}
+			key := fmt.Sprintf("%.10f,%.10f,%.10f", scaledPt.X, scaledPt.Y, scaledPt.Z)
+			if _, exists := vertexMap[key]; !exists {
+				vertexMap[key] = len(vertices)
+				vertices = append(vertices, scaledPt)
+			}
+		}
+	}
+
+	var buf bytes.Buffer
+
+	// Write PLY header
+	buf.WriteString("ply\n")
+	buf.WriteString("format ascii 1.0\n")
+	buf.WriteString(fmt.Sprintf("element vertex %d\n", len(vertices)))
+	buf.WriteString("property float x\n")
+	buf.WriteString("property float y\n")
+	buf.WriteString("property float z\n")
```

Comment:
```
❯ cat mesh_lod_500_June_03_2025_10_50_28.ply
ply
format ascii 1.0
comment Created by Open3D
element vertex 1591
property double x
property double y
property double z
property double nx
property double ny
property double nz
```

shouldn't it be double and not float?

Reply:
also not sure if we want
```
property double nx
property double ny
property double nz
```

Reply:
Ah good point, we want doubles. Brain fart :P

I don't see a need to write normals, we just recalculate them anyway.

Reply:
To be clear, currently the normals in the PLY files are already being discarded and recalculated. This merely takes the additional step of not transmitting them either.

Reply:
oh my bad. 
Do you think that would be worth adding at some point?

Reply:
Meh. Calculating the normal of of three points is really fast. I'm not too concerned about it. I think we ignore that one until or unless it becomes an issue.

The triangles are in the frame of the mesh's pose, so it's not like they have to be recalculated every time we Transform the mesh. If we did then it would be a bigger problem.